### PR TITLE
Segmentation loss

### PR DESCRIPTION
**Pull request by John-HarringtonNZ**

### CODE REVIEW

File: models/cut_model.py [hunk 1]
```diff
@@ -201,6 +234,19 @@ def compute_G_loss(self):
...
         self.loss_G = self.loss_G_GAN + loss_NCE_both
+
+        # SEGMENTATION LOSS
+        if self.segmentation_loss:
+            
+            # Run segmentation model on fake_B
+            seg_fake_B = self.seg_model(self.fake_B.detatch())
+
+            # Run segmentation loss b/w fake_B and GT
+            seg_loss = self.segmentation_criterion(seg_fake_B, self.real_A_seg)
```
Comment (russwong89) [inline models/cut_model.py]: Create a self.loss_seg for plotting purposes and add to self.loss_names

Comment (russwong89) [inline models/cut_model.py]: Should we be ignoring pixels with "unlabeled" class?

File: data/unaligned_dataset.py [hunk 2]
```diff
@@ -58,16 +58,21 @@ def __getitem__(self, index):
...
+        A_seg_img = Image.open(A_seg_path).convert('RGB')
+
         # Apply image transformation
         # For CUT/FastCUT mode, if in finetuning phase (learning rate is decaying),
         # do not perform resize-crop data augmentation of CycleGAN.
         is_finetuning = self.opt.isTrain and self.current_epoch > self.opt.n_epochs
         modified_opt = util.copyconf(self.opt, load_size=self.opt.crop_size if is_finetuning else self.opt.load_size)
         transform = get_transform(modified_opt)
         A = transform(A_img)
+        A_seg = transform(A_seg_img)
```
Comment (russwong89) [inline data/unaligned_dataset.py]: NEed to perform same crop on A and A_seg
https://discuss.pytorch.org/t/torchvision-transfors-how-to-perform-identical-transform-on-both-image-and-target/10606/7
